spec/lib/active_merchant/billing/affirm_spec.rb

```diff
@@ -0,0 +1,41 @@
+require 'spec_helper'
+
+RSpec.describe ActiveMerchant::Billing::Affirm do
+
+  let(:gateway_options) do
+    {
+      public_api_key: "XPSQ3CA7PLN7CJCK",
+      private_api_key: "w9mxkQUryKjTYDqOfSvYTeTGoLIURKpU",
+      test_mode: true
+    }
+  end
+
+  let(:affirm_payment_source) { create(:affirm_checkout, token: "MFLJS57G6BT5OQXN") }
+
+  subject do
+    ActiveMerchant::Billing::Affirm.new(gateway_options)
+  end
+
+  describe "#authorize" do
+    it "with correct payment source" do
+      VCR.use_cassette("valid_authorize") do
```

Comment:
Right, for integration. We don't have any comments indicating why we have a special case here though and the commit message doesn't indicate that it's actually _not possible_ to authorize without going through the affirm interface, and instead the stringent testing for that matter should be done via the feature tests instead.

Also we'll still need to have cases that test errors here.